### PR TITLE
test: replace django_client fixture with the much faster client_logged_in

### PR DESCRIPTION
**Pull request by bruno-fs**
The reality is we don't need a live server for our tests. GOTTA GO FAST.

### CODE REVIEW

File: quipucords/tests/integration/test_smoker.py [hunk 1]
```diff
@@ -206,17 +219,27 @@ def test_deployments_report(  # noqa: PLR0913
...
     ):
         """Compare individual reports with the ones bundled on the tarball endpoint."""
-        details_response = django_client.get(f"reports/{report_id}/details/")
-        assert details_response.ok, details_response.json()
+        details_response = client_logged_in.get(
+            reverse("v1:reports-details", args=(report_id,))
+        )
+        assert (
+            details_response.status_code == status.HTTP_200_OK
+        ), details_response.json()
```
Comment (bruno-fs) [inline quipucords/tests/integration/test_smoker.py:230]: @infinitewarp I'm almost convincing myself to hack django client response so we can keep `response.ok`. it was so elegant... look at this mess! 🤣

**Comment by bruno-fs** [comment]
the file that lose coverage will be removed once @mirekdlugosz get rid of the container integration tests.
I wonder if we can simply ignore this decrease for now 🤔

File: quipucords/tests/api/details_report/test_details_report.py [hunk 2]
```diff
@@ -89,6 +92,6 @@ def test_retrieve_csv(self, sources, django_client):
             "batata,tomate\r\n"
             "\r\n\r\n"
         )
-        assert response.text == expected_csv
+        assert response.content.decode() == expected_csv
```
Comment (bruno-fs) [inline quipucords/tests/api/details_report/test_details_report.py:95]: `.text` is yet another elegant property from requests we are losing

(*urge to hack django client response intensifies*)

File: quipucords/tests/api/reports/test_reports.py [hunk 3]
```diff
@@ -211,25 +225,30 @@ def test_report_tarball_deployments_json(
...
+    details_csv_response = client_logged_in.get(
         reverse("v1:reports-details", args=(deployments_report.report.id,)),
         headers={"Accept": "text/csv"},
     )
-    assert details_csv_response.ok, details_csv_response.text
-    assert details_csv_response.text == tarball_details_csv
+    assert (
+        details_csv_response.status_code == status.HTTP_200_OK
+    ), details_csv_response.content.decode()
+    assert details_csv_response.content.decode() == tarball_details_csv
```
Comment (bruno-fs) [inline quipucords/tests/api/reports/test_reports.py:251]: ugh, my eyes... readability matters a lot 😭

Comment (bruno-fs) [reply quipucords/tests/api/reports/test_reports.py:251]: F readability

**Review by infinitewarp — APPROVED** [review]
LGTM! 👍 

We don't need to address this now, but spotting that test file with _3300+_ lines really makes me want to break it up into multiple more manageable files. Just expanding its diff here on GitHub caused my browser to hang for a while as it loaded everything in.

**Pull request merged**